server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -1296,26 +1313,32 @@ public static String findV2Template(Metadata metadata, String indexName, boolean
     /**
      * Return an ordered list of the name (id) and composable index templates that would apply to an index. The first
      * one is the winner template that is applied to this index. In the event that no templates are matched,
-     * an empty list is returned.
+     * an empty list is returned. If <code>exitOnFirstMatch</code> is true, we return immediately after finding a match.
      */
```

Comment:
I believe we need to elaborate a bit more on how to use this method. What are the trade-offs of `exitOnFirstMatch`. How it should be used etc.